DevCycle.SDK.Server.Local/Api/EventQueue.cs

```diff
@@ -175,22 +178,31 @@ public virtual async Task FlushEvents()
 
         public virtual void QueueEvent(DVCPopulatedUser user, Event @event, BucketedUserConfig config)
         {
+            if (localOptions.DisableEvents)
+                return;
+
+            if (IsOverMaxQueue())
+            {
+                logger.LogWarning("{Event} failed to be queued; events in queue exceed {Max}. Triggering a forced flush", @event, localOptions.MaxEventsInQueue);
+                ScheduleFlushWithDelay();
+                throw new DVCException(new ErrorResponse("Could not add event to queue. Max queue size reached"));
```

Comment:
it seems like we shouldn't be throwing an error here? won't the client then have to worry about catching the `track` call?

Reply:
Well, the concept is that this should be more visible than just a log level; and there's no way of knowing that the add to queue was aborted otherwise.

Reply:
well isn't the log good enough? seems a bit drastic to potentially crash someone's server just because we didn't accept their event

Reply:
how do we tell the application that events are failing to be queued? logs tell the human - but not the application

Reply:
thoughts @squarepegg ?

Reply:
forcing a user to handle this condition when they can't do much about it does feel like an error log is enough.

perhaps a backlog item could be to implement a requeue on flush mechanism?

Reply:
Requeue won't solve the issue of a memory leak when failing to flush - so we won't really have that as a functional solution.

Will change to an error log.